WORKSPACE

```diff
@@ -1,22 +1,30 @@
 workspace(name = "bazel_skylib")
 
-load(":workspace.bzl", "bazel_skylib_workspace")
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+http_archive(
+    name = "bazel_federation",
+    url = "https://github.com/bazelbuild/bazel-federation/archive/130c84ec6d60f31b711400e8445a8d0d4a2b5de8.zip",
+    sha256 = "9d4fdf7cc533af0b50f7dd8e58bea85df3b4454b7ae00056d7090eb98e3515cc",
+    strip_prefix = "bazel-federation-130c84ec6d60f31b711400e8445a8d0d4a2b5de8",
+    type = "zip",
+)
+
+load("@bazel_federation//:repositories.bzl", "bazel_skylib_deps")
+
+bazel_skylib_deps()
```

Comment:
Why do we need to call this function?

Reply:
Right now the function is a no-op since skylib has no dependencies. We can delete that code for now if you want (and add it later once it's actually needed).

Reply:
What's the workflow for adding new dependencies?
First, update the function in the federation, then update skylib's dependency on the federation? Why don't we keep the list of dependencies in Skylib?

Reply:
Yes, that's the process. I know that it's cumbersome, but we wanted to have something that works - improvements to usability will be addressed later.

With the current Starlark-functions-based model I don't see how we can handle transitive dependencies when keeping the dependencies in the projects. All projects would need to define their entire dependency tree, which would lead to a lot of duplicated code across projects.
However, Tony has an idea that involves generating bzl files (if I understood him correctly).